purchase_receipt_percentage/models/purchase_order_line.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_is_zero
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    receipt_percentage = fields.Float(compute="_compute_receipt_percentage", store=True)
+
+    @api.depends("qty_received", "product_qty")
+    def _compute_receipt_percentage(self):
+        for line in self:
+            perc = 100  # 100% by default, in case ordered qty is 0
+            if not float_is_zero(
+                line.product_qty, precision_rounding=line.product_uom.rounding
+            ):
+                perc *= line.qty_received / line.product_qty
+            line.receipt_percentage = min(max(perc, 0), 100)  # 0 <= % <= 100
```

Comment:
Maybe compute this only for lines with product.type in `consu` / `product` ?
I guess it doesn't make sense to compute for services or other kinds, right?
In those cases it should probably default to either 100% or 0%. If it's 0%, then it shouldn't be accounted for in the order's `receipt_percentage` computation

Reply:
In some cases, you might wanna know this data for services as well; for example, a company might buy a 40-hrs training course for an employee, and wants to know how many hours have been effectively used for the training. It'd keep it as it is for the moment, and let project-specific module handle this case if needed.

Reply:
Ah, indeed. I assumed services wouldn't have `qty_received`.. but I've just checked and in fact the user can set it manually.